src/local.rs

```diff
@@ -751,6 +761,42 @@ fn new_staged_upload(base: &std::path::Path) -> Result<(File, PathBuf)> {
     }
 }
 
+fn new_staged_upload_copy_and_append_from(base: &std::path::Path) -> Result<(File, PathBuf, u64)> {
+    let multipart_id = 1;
+    loop {
+        let suffix = multipart_id.to_string();
+        let path = staged_upload_path(base, &suffix);
+
+        let offset = if base.exists() {
+            let offset = std::fs::metadata(base).unwrap().len();
+
+            std::fs::copy(base, &path).map_err(|source| Error::UnableToCopyFile {
```

Comment:
TODO: we might be able to avoid copy to tmp file but write on the single file instead